services/src/test/scala/cromwell/services/metadata/impl/pubsub/PubSubMetadataServiceActorSpec.scala

```diff
@@ -0,0 +1,190 @@
+  }
+
+  /** A variant of PubSubMetadataServiceActor with a GooglePubSubDAO which will always return failure */
+  class FailingToCreateTopicMockPubSubMetadataServiceActor(serviceConfig: Config, globalConfig: Config)
+    extends PubSubMetadataServiceActor(serviceConfig, globalConfig) {
+
+    override def createPubSubConnection(): GooglePubSubDAO = new FailingToCreateTopicMockGooglePubSubDao
+  }
+
+  /** A variant of PubSubMetadataServiceActor which will fail on message publication */
+  class FailToPublishMockPubSubMetadataServiceActor(serviceConfig: Config, globalConfig: Config)
+    extends PubSubMetadataServiceActor(serviceConfig, globalConfig) {
+
+    override def createPubSubConnection(): GooglePubSubDAO = new FailToPublishMockGooglePubSubDao
+  }
+
+  trait MockGooglePubSubDao extends GooglePubSubDAO {
+    override implicit val executionContext = ExecutionContext.global
+
+    override def createTopic(topicName: String): Future[Boolean]
+    override def createSubscription(topicName: String, subscriptionName: String): Future[Boolean]
+    override def publishMessages(topicName: String, messages: Seq[String]): Future[Unit]
+
+    // The following aren't used so leaving them empty
+    override def deleteTopic(topicName: String): Future[Boolean] = ???
+    override def getTopic(topicName: String)(implicit executionContext: ExecutionContext): Future[Option[Topic]] = ???
+    override def deleteSubscription(subscriptionName: String): Future[Boolean] = ???
+    override def acknowledgeMessages(subscriptionName: String, messages: Seq[PubSubMessage]): Future[Unit] = ???
+    override def acknowledgeMessagesById(subscriptionName: String, ackIds: Seq[String]): Future[Unit] = ???
+    override def pullMessages(subscriptionName: String, maxMessages: Int): Future[Seq[PubSubMessage]] = ???
+    override def getPubSubServiceAccountCredential: Credential = ???
+  }
+
+  class SuccessfulMockGooglePubSubDao extends MockGooglePubSubDao {
+    override def createTopic(topicName: String): Future[Boolean] = Future.successful(true)
+    override def createSubscription(topicName: String, subscriptionName: String): Future[Boolean] = Future.successful(true)
+    override def publishMessages(topicName: String, messages: Seq[String]): Future[Unit] = Future.successful(())
+  }
+
+  class FailingToCreateTopicMockGooglePubSubDao extends SuccessfulMockGooglePubSubDao {
+    override def createTopic(topicName: String): Future[Boolean] = Future.failed(new RuntimeException("Unable to create topic"))
+  }
+
+  class FailToPublishMockGooglePubSubDao extends SuccessfulMockGooglePubSubDao {
+    override def publishMessages(topicName: String, messages: Seq[String]): Future[Unit] = Future.failed(new RuntimeException("sorry charlie"))
+  }
+
+  // This doesn't include a project so should be a failure
+  val emptyConfig = ConfigFactory.parseString(
+    """
+      |
+    """.stripMargin
```

Comment:
A little verbose for what could be `""` (or in this case `"\n\n  \n"`)

Reply:
Or apparently `ConfigFactory.empty()`